### PR TITLE
Group agency search results by letter and adds number of results. 

### PR DESCRIPTION
**Pull request by ramirezg**
Solves 18F/foia-hub/issues/464
Still needs some template work, posting this so we can discuss the structure.

### CODE REVIEW

File: foia_hub/templates/contacts/index.html [hunk 1]
```diff
@@ -13,26 +13,34 @@
...
-                {% else %}
-                  Click to find out more about this agency.
-                {% endif %}
-              </p>
-            </div>
-          </li>
+    {% endif %}
+      <h3>Results: {{agencies|length}}</h3>
+      <p>
+          {% for letter in agencies|groupby('name.0') %}
```
Comment (ramirezg) [inline foia_hub/templates/contacts/index.html]: Thoughts on using two loops for adding anchor tags later?

Comment (khandelwal) [inline foia_hub/templates/contacts/index.html]: I don't understand what you're asking.

Comment (cmc333333) [inline foia_hub/templates/contacts/index.html]: You could set the results as a variable, e.g. `{% set grouped_agencies = agencies|groupby('name.0') %}` to make sure the logic is consistent, but you'd still be iterating twice. That seems okay to me. If the first loop ever gets too complicated (and you want to keep both loops visible in the same screen), consider a [macro](http://jinja.pocoo.org/docs/dev/templates/#macros).

Comment (khandelwal) [inline foia_hub/templates/contacts/index.html]: Why do we need two loops to add anchor tags?

Comment (cmc333333) [inline foia_hub/templates/contacts/index.html]: Looks like:
```
A C R S

A
AStar Agency

C
Candidate Agency

R
Red Agency

S
Super Agency
```

The listing at the top is a separate loop from the listing at the bottom. Hypothetically, you could combine them into a single loop using concatenated templates but that sounds very ugly.

**Comment by khandelwal** [comment]
You still need to add anchor tags to this.

**Comment by ramirezg** [comment]
@vz3 Let me know if this works. 

screen shot 2015-02-04 at 12 40 07 pm

**Comment by vz3** [comment]
Here's the comp.

For "All agencies", need to not display number of results. We're not currently doing pagination, so ignore "Display 1-10" for now. Will I be able to stylize the A in the above example? I'd need to add something like:
```<div class="letter">
<h2>A</h2><h4>Back to top</h4>
</div>```
![searchresults1](https://cloud.githubusercontent.com/assets/582918/6047945/d6dd5746-ac7a-11e4-9ba4-df2fbff32bd0.png)
![searchresults2](https://cloud.githubusercontent.com/assets/582918/6047946/d6e12be6-ac7a-11e4-881a-a20ac9c925fc.png)

**Comment by ramirezg** [comment]
@vz3 ok let me know if this works.

**Pull request merged**